pkg/autopilot/download/downloader.go

```diff
@@ -37,52 +40,58 @@ type Config struct {
 }
 
 type downloader struct {
-	config       Config
-	logger       *logrus.Entry
-	httpResponse *grab.Response
+	config Config
 }
 
 var _ Downloader = (*downloader)(nil)
 
-func NewDownloader(config Config, logger *logrus.Entry) Downloader {
+func NewDownloader(config Config) Downloader {
 	return &downloader{
 		config: config,
-		logger: logger.WithField("component", "downloader"),
 	}
 }
 
-// Start begins the download process, starting the downloading functionality
-// on a separate goroutine. Cancelling the context will abort this operation
-// once started.
-func (d *downloader) Download(ctx context.Context) error {
-	// Setup the library for downloading HTTP content ..
-	dlreq, err := grab.NewRequest(d.config.DownloadDir, d.config.URL)
-
-	if err != nil {
-		return fmt.Errorf("invalid download request: %w", err)
-	}
+// Performs the download process.
+func (d *downloader) Download(ctx context.Context) (err error) {
+	var targets []io.Writer
 
 	// If we've been provided a hash and actual value to compare with, use it.
+	var expectedHash []byte
 	if d.config.Hasher != nil && d.config.ExpectedHash != "" {
-		expectedHash, err := hex.DecodeString(d.config.ExpectedHash)
+		expectedHash, err = hex.DecodeString(d.config.ExpectedHash)
 		if err != nil {
 			return fmt.Errorf("invalid update hash: %w", err)
 		}
+		targets = append(targets, d.config.Hasher)
+	}
 
-		dlreq.SetChecksum(d.config.Hasher, expectedHash, true)
+	// Set up target file for download.
+	target, err := file.AtomicWithTarget(filepath.Join(d.config.DownloadDir, "download")).Open()
```

Comment:
hmm, doesn't this mean that `Download` func can only download one file at a time as the tmp file is always named `download`?

Reply:
No, it's using this as the base name only.